glossary.md

```diff
@@ -0,0 +1,241 @@
+[L2 derivation inputs][deriv-inputs] available on L1. This is done by its
+[rollup driver] component.
+
+The rollup node is also responsible for [block gossip], the process of
+transmitting the L2 blocks it derives over a peer-to-peer network.
+
+- cf. [Rollup Node Specification](/rollup-node.md)
+- cf. [Block Gossip Specification](TODO)
+
+> **TODO LINK** block gossip spec
+
+## Rollup Driver
+[rollup driver]: /glossary.md#rollup-driver
+
+The rollup driver is the [rollup node] component responsible for [deriving the
+L2 chain][derivation] from the [L2 derivation inputs][deriv-inputs] available on
+L1.
+
+## L2 Chain Derivation
+[derivation]:  /glossary.md#L2-chain-derivation
+
+A process that reads [L2 derivation inputs][deriv-inputs] from L1 in order to
+derive the L2 chain.
+
+cf. [L2 Chain Derivation (in Rollup Node
+Specification)](/rollup-node.md#l2-chain-derivation)
+
+## L2 Derivation Inputs
+[deriv-inputs]: /glossary.md#l2-chain-derivation-inputs
+
+This term refers to data that is found in L1 blocks and is read by the [rollup
+node] to construct [payload attributes].
+
+Chain derivation attributes include:
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
+- [deposits]
+
+## Payload Attributes
+[payload attributes]: /glossary.md#payload-attributes
+
+This term refers to data that can be derived from [L2 chain derivation
+inputs][deriv-inputs] found on L1, which are then passed to the [execution
+engine] to construct L2 blocks.
+
+"Payload attributes" is a term that originates and is specified in the [Ethereum
+Engine API specification][engine-api].
+
+In addition to the [specified][engine-api] `timestamp`, `random`and
+`suggestedFeeRecipient` fields, we add a field to specify the transactions that
+the execution engine must include in the block and execute.
+
+> **TODO** The transaction field must be specified.
+
+Payload attributes were historically called "L2 block inputs" and you might
+still hear some people using this term.
+
+## L1 Attributes Transaction
+[l1-attributes-tx]: /glossary.md#l1-attributes-transaction
+
+A transaction with an Optimistic-Ethereum-specific transaction type, that is
+used to register the L1 block attributes (number, timestamp, ...) on L2.
+
+The L1 attributes for a given L1 block can be read on L2 from the [L1 Attributes
+Precompiled Contract][l1-attr-precompile].
+
+cf. [L1 attributes transaction format](/rollup-node.md#payload-transaction-format)
+(in the section on [payload attributes])
+
+> **TODO** We might want to move this the format spec to the execution engine.
+
+## L1 Attributes Precompiled Contract
+[l1-attr-precompile]: /glossary.md#l1-attributes-precompiled-contract
+
+A [precompiled contract][precompile] on L2 that can be used to retrieve the L1
+block attributes of L1 blocks with a given block number or a given block hash.
+
+cf. [L1 Attributes Precompiled Contract Specification](TODO)
+
+> **TODO LINK** L1 attributes precompiled contract spec
```

Comment:
It looks like a precompile, but we don't need it to be one, and I would prefer not to change geth when we do not have to. In this case we want upgradeability, storage, and updates through transactions. Reusing the EVM smart contracts, state storage, and deposit transactions covers everything we need. No need for a precompile.

Reply:
For the sake of brevity and concentrating the discourse, could we avoid making the same comment in three different places ;)